test/e2e/node_feature_discovery_test.go

```diff
@@ -747,24 +815,28 @@ func waitForNfdNodeAnnotations(cli clientset.Interface, expected map[string]stri
 
 type k8sLabels map[string]string
 
-// waitForNfdNodeLabels waits for node to be labeled as expected.
-func waitForNfdNodeLabels(cli clientset.Interface, expected map[string]k8sLabels) error {
+// checkForNfdNodeLabels waits and checks that node is labeled as expected.
+func checkForNodeLabels(cli clientset.Interface, expectedNewLabels map[string]k8sLabels, oldNodes []corev1.Node) error {
+
 	poll := func() error {
 		nodes, err := getNonControlPlaneNodes(cli)
 		if err != nil {
 			return err
 		}
 		for _, node := range nodes {
-			labels := nfdLabels(node.Labels)
-			nodeExpected, ok := expected[node.Name]
+			nodeExpected, ok := expectedNewLabels[node.Name]
 			if !ok {
 				nodeExpected = k8sLabels{}
-				if defaultExpected, ok := expected["*"]; ok {
+				if defaultExpected, ok := expectedNewLabels["*"]; ok {
 					nodeExpected = defaultExpected
 				}
 			}
-			if !cmp.Equal(nodeExpected, labels) {
-				return fmt.Errorf("node %q labels do not match expected, diff (expected vs. received): %s", node.Name, cmp.Diff(nodeExpected, labels))
+
+			oldLabels := getNodeLabels(oldNodes, node.Name)
+
+			maps.Copy(maps.Clone(oldLabels), nodeExpected)
```

Comment:
This doesn't look correct now. Shouldn't it be smth like 

```suggestion
			expectedNewLabels := maps.Copy(maps.Clone(oldLabels), nodeExpected)
```

Reply:
```golang
func maps.Copy(dst map[string]string, src k8sLabels)
```
Actually, Copy function doesn't return anything, it copies from src into a destination.

Reply:
True, my bad